scarb/src/ops/package.rs

```diff
@@ -246,6 +277,32 @@ fn prepare_archive_recipe(pkg: &Package, opts: &PackageOpts) -> Result<ArchiveRe
         contents: ArchiveFileContents::OnDisk(pkg.manifest_path().to_owned()),
     });
 
+    if pkg.is_cairo_plugin() && !pkg.is_builtin() {
+        // Package crate with Cargo.
+        package_package(pkg, opts, ws)?;
+
+        // Add normalized Cargo.toml file.
+        recipe.push(ArchiveFile {
+            path: CARGO_MANIFEST_FILE_NAME.into(),
+            contents: ArchiveFileContents::OnDisk(
+                pkg.target_path(ws.config())
+                    .into_child("package")
+                    .into_child(get_crate_archive_basename(
+                        pkg.root().join(CARGO_MANIFEST_FILE_NAME),
```

Comment:
Note this points to unnormalized manifest.

Reply:
Updated the `get_crate_archive_basename` logic to extract package name from Cargo.toml and package version from `cargo metadata`